### DISCUSSION TITLE
Forge/Bukkit Hybrid Category?

### DISCUSSION THREAD

**Discussion by LeStegii** [modrinth/.github#36] [Other discussions]
Hello!
Before I start, this isn't meant to be a feature request, but more of a question/discussion about this topic.

Currently, the modrinth website has many different categories for plugins reaching from basic CraftBukkit to Purpur (and SpongeForge). 

But aside from those softwares, there is a category of softwares which doesn't really fit into one of these. I'm talking about hybrids like Arclight, Magma, Mohist, Crucible (...) which implement the bukkit API into Forge. While many mod/plugin developers advise against this kind of software, it had its fair place in modded communities like Pixelmon for years.

Most of these softwares also offer advanced api for allowing plugins to interact with mods or vice versa, making some plugins work in this enviroment only. Until now, these plugins had to be uploaded to GitHub or other sites, but there wasn't a central place where you could download such plugins since they don't really fit into sites like SpigotMC.org.

Since there aren't that many plugins yet, I'm unsure if a whole dedicated loader category would be fitting, but on the other hand, I doubt there are many plugins on Modrinth which *only* run on Purpur for example.

I'm really interested in what you think about this idea.
If you're interested in a list of plugins which would fit this category let me know.

Have a nice day,
@UeberallGebannt

**Comment by Prospector** [comment]
I'm not familiar with this, but if it implements the Bukkit API on forge, does that not just mean it loads Bukkit plugins? Why would it need a separate category and not just be in the Bukkit category?

**Reply by LeStegii** [reply]
Yes, they do, but like I said: Most of them offer advanced api for things like communication between mods and plugins, allowing for stuff like syncing Pixelmon economy with Vault. This obviously wouldn't work on plain bukkit.

It's similar to Purpur: Purpur also implements the Paper API, but has extended API which means there could be plugins only running on Purpur but not on Paper (or Spigot/Bukkit).

**Reply by Prospector** [reply]
Could you maybe send a list of links to each of these projects that has their own API with?

**Reply by LeStegii** [reply]
Yes of course!

Here are some of the mentioned hybrids, there are a few more (inactive) ones I think, but these are the main ones currently used by the community:
https://magmafoundation.org for 1.12, 1.18
https://github.com/IzzelAliz/Arclight for 1.16, 1.18 and 1.19
https://mohistmc.com/ for 1.12, 1.16 and 1.19
https://cruciblemc.github.io/ for 1.7.10

I also have a list of some example plugins which would require such a hybrid to work:
https://www.spigotmc.org/resources/pixelmoneconomybridge.97889/
https://www.spigotmc.org/resources/bukkitpixelmonplaceholders.105174/
https://github.com/UknownGino/ModdedIntegration

**Comment by Andre601** [comment]
I for my part would discourage adding a hybrid category.
Forge and bukkit are two different API types which are incompatible with each other.

These hybrid servers just smash them together without any thought in the world about plugin/mod compatability.

Such servers shouldn't be supported by giving them their own category.

If you want mods with plugins, use SpongeForge, as its plugin API was made with mod support in mind.

**Reply by LeStegii** [reply]
I totally see your point, the APIs have not been made to work with each other, but still, there is a relatively big community around these, as they are used as the base for Pixelmon servers quite often and are quite stable at this point. I'm not a part of the pixelmon community though, so I can only tell what I see from the outside.

Since this one is personal I wouldn't count it as an argument for hybrids, but I had less problems with mods working than on SpongeForge back in the days of 1.12.2. (Edit: I heard some people say the exact opposite also)

**Reply by Andre601** [reply]
Just because a community is big should a bad decition (Merging 2 incompatible APIs) be supported with a category on Modrinth?

I really can't see your logic here. It doesn't really matter if the APIs are "stable" (They aren't), it's still doing something that shouldn't be done that way, and that already has a good alternative available to use (SpongeForge).

Hybrids like Mohist also put some morally bad things on themself by doing stuff like replacing a plugin jar with one they modified to "fix" their own mistakes ([Source](https://essentialsx.net/do-not-use-mohist.html)).
